authentication/http.go

```diff
@@ -39,23 +37,25 @@ func WithTenant(next http.Handler) http.Handler {
 
 // WithTenantID returns a middleware that finds the tenantID using the tenant
 // from the URL parameters and adds it to the request context.
-func WithTenantID(tenantIDs map[string]string) Middleware {
+func WithTenantID(tenantName, tenantID string) Middleware {
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			tenant := chi.URLParam(r, "tenant")
+			queryValues := r.URL.Query()
```

Comment:
The tenant name is part of the path, not the query

Reply:
Ack. This is tricky. 
Current implementation : {} notation around {tenant} is being used for Chi router to pickup the tenant name from the URL and 'automatically' include as a Query parameter for downstream functions to figure out the tenant name. 
Proposed change: {} notation of Chi is not compatible for separate Mounts() for each tenant. So explicitly adding tenant name as a query parameter to avoid bigger changes in the downstream functions.

Reply:
Hmm this sounds funny, so this is changing the API to no longer use path parameters and instead move the tenant parameter to the query?

Reply:
this change explicitly adds query param ('tenant') which previously automatically added by Chi router with {tenant} at the query path.

Reply:
> which previously automatically added by Chi router with {tenant} at the query path.

I need to agree with Lucas here, I think it previously was added as `path` not as `query`, no?

See it being used in test: https://github.com/observatorium/api/blob/master/test/integration.sh#L142 - it was used in path. With this new implementation I assume this test would fail?

Reply:
In any case, this is certainly an API change: we used path parameters before and now use path parameters AND query parameters. I think this is not a good interface as it makes things difficult for clients. Not to mention it's confusing. What happens if you specify the url `/API/V1/foo/graph?tenant=bar`? What tenant is actually used?

Reply:
The query-param is not expected as part of the call (automatically added by Chi) for picking up the tenant-name (gets added to the context). 
@squat I did not test the possible scenario you have outlined. 
With adding tenant name to the request's context, removed code (prior, changes with this PR) related to adding / reading query-param for tenant-name.